src/find_service/server.rs

```diff
@@ -15,73 +12,54 @@ fn convert_system_time_error(time_error: time::SystemTimeError) -> io::Error {
     io::Error::new(io::ErrorKind::Other, time_error)
 }
 
-type PublisherStore = Arc<RwLock<HashMap<String, proto::Registration>>>;
+#[derive(Debug, Clone)]
+pub struct PublisherNotFoundError;
+
+impl fmt::Display for PublisherNotFoundError {
+    fn fmt(&self, f: &mut fmt::Formatter) -> fmt::Result {
+        write!(f, "invalid first item to double")
+    }
+}
+
+pub trait PublisherStore: Send + Sync + Clone + 'static {
+    fn insert_publisher(&self, publisher_name: &String, registration: proto::Registration);
+    fn remove_publisher(&self, publisher_name: &String) -> Result<(), PublisherNotFoundError>;
+    fn remove_publishers(&self, publisher_names: &Vec<String>);
```

Comment:
This should probably provide a default implementation
```suggestion
    fn remove_publishers(&self, publisher_names: &Vec<String>) {
        for publisher in publisher_names {
            self.remove_publisher(publisher);
        }
    }
```

Reply:
Another debate I have is `&Vec<T>` vs `&[T]`. it seems like there is not much difference for our purposes. and `&[T]` has less baggage.